src/ets/scala/io/iohk/ethereum/ets/blockchain/BlockchainSuite.scala

```diff
@@ -0,0 +1,94 @@
+  val ignoredTests: Map[String, Set[String]] = Map(
+    "bcForgedTest/bcForkUncle" -> Set("ForkUncle"),
+    "bcForkStressTest/ForkStressTest"  -> Set.empty,
+    "bcInvalidHeaderTest/GasLimitHigherThan2p63m1"  -> Set.empty,
+    "bcMultiChainTest/CallContractFromNotBestBlock" -> Set.empty,
+    "bcMultiChainTest/ChainAtoChainB_blockorder1" -> Set.empty,
+    "bcMultiChainTest/ChainAtoChainB_blockorder2" -> Set.empty,
+    "bcMultiChainTest/ChainAtoChainBtoChainA" -> Set.empty,
+    "bcTotalDifficultyTest/lotsOfBranches" -> Set.empty,
+    "bcTotalDifficultyTest/lotsOfBranchesOverrideAtTheMiddle"  -> Set.empty,
+    "bcTotalDifficultyTest/lotsOfLeafs"  -> Set.empty,
+    "bcTotalDifficultyTest/sideChainWithMoreTransactions" -> Set.empty,
+    "bcTotalDifficultyTest/uncleBlockAtBlock3afterBlock4" -> Set.empty,
+    "TransitionTests/bcFrontierToHomestead/blockChainFrontierWithLargerTDvsHomesteadBlockchain"  -> Set.empty,
+    "TransitionTests/bcFrontierToHomestead/blockChainFrontierWithLargerTDvsHomesteadBlockchain2"  -> Set.empty
+  )
+
+  override def run(testName: Option[String], args: Args): Status = {
+    val options = TestOptions(args.configMap)
+    val scenarios = BlockchainScenarioLoader.load("ets/BlockchainTests/", options)
+
+    scenarios.foreach { group =>
+      group.name - {
+        for {
+          (name, scenario) <- group.scenarios
+          if options.isScenarioIncluded(name)
+        } {
+          name in new ScenarioSetup(scenario) {
+            if (unsupportedNetworks.contains(scenario.network)) {
+              cancel(s"Unsupported network: ${scenario.network}")
+            } else if (!supportedNetworks.contains(scenario.network)) {
+              fail(s"Unknown network: ${scenario.network}")
+            } else if (isCanceled(group.name, name)){
+              cancel(s"Test: $name in group: ${group.name} not yet supported")
+            } else {
+              log.info(s"Running test: ${group.name}/$name")
+              runScenario(scenario, this)
+            }
+          }
+        }
+      }
+    }
+
+    runTests(testName, args)
+  }
+
+  private def isCanceled(groupName: String, testName: String): Boolean =
+    ignoredTests.get(groupName).isDefined && (ignoredTests(groupName).contains(testName) || ignoredTests(groupName).isEmpty)
+
+  private def runScenario(scenario: BlockchainScenario, setup: ScenarioSetup): Unit = {
+    import setup._
+
+    val genesisBlock = loadGenesis()
+
+    loadInitialWorld()
+
+    val blocksToProcess = getBlocks(scenario.blocks)
+
+    val invalidBlocks = getBlocks(getInvalid)
+
+    val newBlocks: (Seq[NewBlock], Seq[BlockExecutionError]) = processBlocks(blocksToProcess)
+
+    // If there is more block execution errors than expected invalidblocks, it means we rejected block which should
+    // pass validations, and the final state will not be correct
+    if(newBlocks._2.size > invalidBlocks.size)
```

Comment:
> Maybe it's not considered in the BlockchainSuite scenarios, but maybe we should also consider the case that newBlocks._2.size < invalidBlocks.size?
I'm not sure how this scenario is handled but maybe the test should fail in case this size comparison is invalid?

Was this comment from @ntallar addressed?

Reply:
ahh sorry @ntallar if forgot to properly address it, during comments barrage from Radek. Actually I tested it now and I think it is worth having assertion `newBlocks._2.size > invalidBlocks.size` as it helps to catch to false positives (both connected to branching and forking)